.crawler/src/known_network.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use parking_lot::RwLock;
+use std::{
+    collections::{HashMap, HashSet},
+    net::SocketAddr,
+};
+use time::OffsetDateTime;
+
+use crate::connection::{nodes_from_connections, Connection};
+
+// Purges connections that haven't been seen within this time (in hours).
+const STALE_CONNECTION_CUTOFF_TIME_HRS: i64 = 4;
+
+// The interval for revisiting connections.
+const CRAWL_INTERVAL_MINS: i64 = 10;
+
+/// Node information collected while crawling.
+#[derive(Debug, Default, Clone)]
+pub struct NodeMeta {
+    last_height: u32,
+    // Set on disconnect.
+    last_crawled: Option<OffsetDateTime>,
+
```

Comment:
I know that we'll also have this info available in connections, but we might also want to keep the node's address here, we might need it in the future for more detailed analysis